### ISSUE TITLE
Allow dragging in "fill" mode in labels

### ISSUE THREAD

**Issue by jni** [napari/napari#1235]
## 🚀 Feature

When painting many labels the same color (e.g. filling in background, or bigger objects from superpixels), if one already has reasonable superpixels, it's much easier to drag and roughly paint over the whole object than to click on each individual pixel.

**Comment by sofroniewn** [comment]
Can you describe this a little more actually @jni I'm not sure i fully understand enough to know what is requested.

Say i have fill bucket tool highlighted, and then i click on a pixel, right now it fills all contiguous pixels of that label with the new label.

If I click and start dragging what do you want to happen? Do you want it to start moving that contiguous component around? Should it "paint" with that new label into new pixels? Should it fill in where it used to be with the background label or should it only grow? Should it actually only modifiy pixels when you release? That would be pretty cool, filling in background with where it used to be?

Screenshots might help too

**Comment by jni** [comment]
dragging should be equivalent to clicking on each pixel in the drag path. In other words, it should be like moving and clicking fill many times over.

**Comment by jni** [comment]
(which is what you need to do now to paint a superpixel image.)

**Comment by jni** [comment]
Here's some code to play around with:

```python
import napari
from skimage import data, segmentation

astro = data.astronaut()
segmented = segmentation.slic(
    astro, compactness=20, sigma=1, start_label=1
)

with napari.gui_qt():
    v = napari.Viewer()
    imglayer = v.add_image(astro)
    seglayer = v.add_labels(segmented)
    seglayer.mode = 'fill'
```

Now we define the foreground as the astronaut's face and background as everything else. We want to fill the background superpixels with 0 and the foreground superpixels with 1. I guarantee you'll be tired of clicking before you're done with the background and you'll wish you could just drag across it. :joy:

**Comment by jni** [comment]
(based on a true story =P)

**Comment by VolkerH** [comment]
I like this example. From the docs, it appears that skimage can do slics only on colour images. With superpixels on greyvalue images, this would allow for a "magic wand"-like action which should greatly speed up annotations and accuracy around edges.

**Comment by sofroniewn** [comment]
I get it now!! Thanks for the explanation @jni. Agreed clicking got quite tiring!! If I had been able to click and then fill while dragging that would be very nice. Here's a video of me clicking each superpixel:

![fill_drag](https://user-images.githubusercontent.com/6531703/82682427-17b8a300-9c04-11ea-8ff8-301facc3dbd2.gif)

**Comment by liu-ziyang** [comment]
Opened PR https://github.com/napari/napari/pull/1299/files, I tested the astronaut case, was able to drag to fill the areas.

However I am a bit concerned in general about filling with dragging. Maybe just a personal impression, but filling has been a relatively dangerous operation to me because the outcome may not be immediately obvious to the user, and can be potentially very damaging to what the user is trying to do because the fill can potentially fill a lot bigger area than expected. Even though undo mitigates some of the mental pressure on the user, and allow people to try freely, I am still traumatized by a few "Shit!" moments when I messed up my fills...

**Comment by sofroniewn** [comment]
Definitely having the undo functionality here is useful!!

**Comment by jni** [comment]
Oh man I disagree totally about the risks. Maybe I have steady hands but I tried this just now and it's just *awesome*, so nice to paint those fragments using #1299! Thanks @ziyangczi! :tada: 

I agree with @sofroniewn's review, in particular with the clipping. But all around, :tada:!

**Comment by liu-ziyang** [comment]
shall we close this issue now?

**Comment by sofroniewn** [comment]
Closed by #1299
